src/Symfony/Component/Debug/DebugClassLoader.php

```diff
@@ -128,6 +128,18 @@ public static function disable()
         }
     }
 
+    /**
+     * Finds a file by class name.
+     *
+     * @param string $class
+     *
+     * @return mixed Depends of the wrapped class loader "findFile" method implementation
+     */
+    public function findFile($class)
+    {
+        return $this->isFinder ? $this->classLoader[0]->findFile($class) : false;
```

Comment:
`return $this->isFinder ? $this->classLoader[0]->findFile($class) ?: null : null;`

Reply:
What is the purpose of not returning directly the result of the "findFile" method ?

By checking what it returns, we assume the implementation of the method. I guess that technically, the "findFile" method could return the integer 0 when the file has not been found.

Reply:
i think the idea here is `DebugClassLoader::findFile()` should be defined API, regardless it's implementation. But then we might want to cast truthy values to string beforehand.

:+1: for string|null :)

Reply:
Thank you, I think I see what you mean : the `DebugClassLoader::findFile()` method is a deliberate implementation on our side. We don't try to hide the `DebugClassLoader` to 3rd parties calling this method.

Reply:
What about `return $this->isFinder ? (\is_string($file = $this->classLoader[0]->findFile($class)) ? $file : null) : null;` ?

I guess a file path can only be defined by a string.

Casting truthy values to string can lead to errors if the returned value by wrapped class loader is an array or an object.

Reply:
> I guess a file path can only be defined by a string.

perhaps cast objects implementing __toString at least (SplFileInfo & co)

Reply:
let's keep things simple, no need to be compatible with theoretical implementations that don't exist - and that we don't care being "compatible" with.
I stick to my initial proposal.

Reply:
and not needed actually :D https://3v4l.org/nclRV

Reply:
In this case we cannot add the `?string` return type safely. Should we still do it because "we don't care" ?

Reply:
works? https://3v4l.org/2nsOW

whenever strict types are enabled it just crashes, then we dont care anymore: https://3v4l.org/b38le

Reply:
Strict types are not enabled for this file, @fancyweb is right.

> we cannot add the ?string return type safely

yes we can. A TypeError exception is the proper of reporting a badly wrapped "finder". Wich is theory again.

Reply:
https://3v4l.org/Sd1Ut : crashes without strict type with an object not implementing `__toString()`.

But I guess we don't care. Let's not waste more time on this. This `findFile"` method will almost never be called and `?string` will probably work 99.9% of time.

Reply:
that's not a crash: that's a TypeError, which is a legit behavior and what we also want here :)

Reply:
Sorry I wrote my previous comment before reading yours. You had already explained it :) I now repushed updated code.